Xamarin.Forms.Core.UnitTests/GridTests.cs

```diff
@@ -271,28 +454,85 @@ protected override SizeRequest OnMeasure(double widthConstraint, double heightCo
 			}
-			
+
+			protected override SizeRequest OnMeasure(double widthConstraint, double heightConstraint)
+			{
+				double minWidth = 10.2857142857143;
+
+				_counter += 1;
+
+				switch (_counter % 6)
+				{
+					case 1:
+						return new SizeRequest(
+							new Size(375.619047619048, 673.904761904762),
+							new Size(minWidth, 673.904761904762));
+					case 2:
+						return new SizeRequest(
+							new Size(411.428571428571, 575.619047619048),
+							new Size(minWidth, 575.619047619048));
+					case 3:
+						return new SizeRequest(
+							new Size(336.380952380952, 755.809523809524),
+							new Size(minWidth, 755.809523809524));
+					case 4:
+						return new SizeRequest(
+							new Size(375.619047619048, 673.904761904762),
+							new Size(minWidth, 673.904761904762));
+					case 5:
+						return new SizeRequest(
+							new Size(313.142857142857, 772.190476190476),
+							new Size(minWidth, 772.190476190476));
+					case 0:
+						return new SizeRequest(
+							new Size(313.142857142857, 772.190476190476),
+							new Size(minWidth, 772.190476190476));
+				}
+
+				throw new Exception("This shouldn't happen, unless we make measure/layout more or less efficient and " +
+					"OnMeasure isn't called 6 times during this test.");
+			}
+		}
+
+		class HeightBasedOnTextLengthLabel : TestLabel
+		{
+			public double DesiredHeight(double widthConstraint) 
+			{ 
+				return (Text.Length / widthConstraint) * 200;
```

Comment:
should this be widthConstraint?

```suggestion
				return (Text.Length / widthConstraint) * widthConstraint;
```

Reply:
If `Text.Length` == x and `widthConstraint` == y, then that would be 
`(x / y) * y`
which is 
`x * y / y`
which is `x`. So I'd just be returning `Text.Length`.

Reply:
Just a heads up that I tried to switch the parent container from Grid to StackLayout in order to work around this bug, but I'm still getting labels with text being cut off. Maybe it's calculating the cell size wrong?

Reply:
@modplug can you log a new issue?6